base_wamas_ubl/lib/wamas/ubl2wamas.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/python3
+
+import getopt
+import logging
+import sys
+from datetime import date, datetime
+
+import xmltodict
+from dateutil.parser import parse
+from dotty_dict import Dotty
+
+from .wamas2ubl import MappingDict
+
+_logger = logging.getLogger("wamas2ubl")
+
+# TODO: Find "clean" way to manage imports for both module & CLI contexts
+try:
+    from .utils import *  # noqa: F403
+    from .wamas_grammar import ausk, ausp, weak, weap  # noqa: F401
+except ImportError:
+    from utils import *  # noqa: F403
+    from wamas_grammar import ausk, ausp, weak, weap  # noqa: F401
+
+
+LST_FIELD_UNIT_CODE = ["HostEinheit"]
+
+
+MAPPING_UBL_TO_WAMAS = {
+    # DespatchLine/DeliveredQuantity[unitCode]
+    # https://docs.peppol.eu/poacc/upgrade-3/codelist/UNECERec20/
+    "unitCode": MappingDict(
+        {
+            "XBQ": "BOT",  # plastic bottle
+            "XBX": "BOITE",  # box
+            "LTR": "LITRE",  # litre
+            "XBO": "PET",  # glass bottle
+            "X4B": "TETRA",  # tetra pack
+            "": False,  # undefined,
+        }
+    )
+}
+
+
+def _set_string(val, length, dp):
```

Comment:
could you add a `test_ubl2wamas.py` that tests for expected values of these helpers?